In [108]:
import pandas as pd
import numpy as np

In [109]:
df=pd.read_csv("C:\\Users\\sb193\\OneDrive\\Documents\\salay_dataset.csv")

In [110]:
df.head()

YearsExperience  Salary
0              1.1   39343
1              1.3   46205
2              1.5   37731
3              2.0   43525
4              2.2   39891

# What will be the salary of an individual having 5years of experiece

For this we take  YearsExperience as X column and Salary as Y column

In [113]:
n=len(df)
n

30

In [114]:
x=(n*((df["YearsExperience"]*df["Salary"]).sum())-df["YearsExperience"].sum()*df["Salary"].sum())
x

66212484.0

In [115]:
y=(n*(df["YearsExperience"]**2).sum()-(df["YearsExperience"].sum())**2)
y

7006.639999999999

In [116]:
slope=x/y
slope

9449.962321455077

In [117]:
intercept=(df["Salary"].sum()-slope*df["YearsExperience"].sum())/(n)
intercept

25792.200198668685

In [118]:
Y=round((slope*5+intercept),2)
Y

73042.01

The salary of an individual having 5years of experiece is 73042.01

# What will be the experience of an individual having 50k salary

For this we take Salary as X column and YearsExperience as Y column

In [122]:
a=(n*((df["Salary"]*df["YearsExperience"]).sum())-df["Salary"].sum()*df["YearsExperience"].sum())
a

66212484.0

In [123]:
b=(n*(df["Salary"]**2).sum()-(df["Salary"].sum())**2)
b

653849335560

In [124]:
slope=a/b
slope

0.00010126565922605279

In [125]:
intercept=(df["YearsExperience"].sum()-slope*df["Salary"].sum())/(n)
intercept

-2.383160564824357

In [126]:
X=round((slope*50000+intercept),2)
X

2.68

The experience of an individual having 50k salary is 2.68
